recipes/doxygen/all/conanfile.py

```diff
@@ -77,6 +86,20 @@ def package(self):
         cmake = self._configure_cmake()
         cmake.install()
 
+    def package_id(self):
+        del self.info.settings.arch_build
+        del self.info.settings.os_build
+        del self.info.settings.compiler
+
+        # Doxygen doesn't make code. Any package that will run is ok to use.
```

Comment:
The removal of compiler is probably correct since Doxygen is a tool. 

I don't follow why you adding arch_build and os_build (both settings which could be seen as deprecated) and compatible packages. I can't execute Doxygen for another arch than my host has

Reply:
Remove it 

https://github.com/conan-io/conan-center-index/blob/master/docs/faqs.md#why-installer-packages-remove-some-settings-from-their-package-id

Reply:
The compatible packages part is so that we don't have to build Doxygen many times to be able to build our various configurations. The old `doxygen_installer` recipe depended on `arch_build` and `os_build` for selecting the package, but I get that we don't want that for the _primary_ package any more.

- It looks at `arch_build`, because, for instance, it's okay to use a 64-bit Doxygen tool on 64-bit Windows if we're building 32-bit packages with `-s arch=x86`.
- One of our engineers pointed out that doing the same for `os_build` covers cross-building.
- It's totally ok to use a Release build of Doxygen if building a Debug product and there isn't a Debug Doxygen around.
- The [discussion](https://github.com/conan-io/conan-center-index/issues/3157#issuecomment-714565227) in #3157 suggests using compatible_packages.